editor/src/core/workers/parser-printer/parser-printer-transpiling.ts

```diff
@@ -243,17 +248,72 @@ export function insertDataUIDsIntoCode(
   }
 }
 
+function getAbsoluteOffsetFromFile(
+  lines: string[],
+  { line, column }: { line: number; column: number },
+): number {
+  if (line >= lines.length) {
+    return -1
+  }
+
+  let offset = 0
+  for (let i = 0; i < line - 1; i++) {
+    // Add 1 for newline character
+    offset += lines[i].length + 1
+  }
+  offset += column
+  return offset
+}
+
+function applySteganographyPlugin(
+  sourceFileName: string,
+  mapToUse: RawSourceMap,
+  fileLines: string[],
+) {
+  return () => ({
+    visitor: {
+      StringLiteral(path: any) {
+        if (path.node.loc == null) {
+          return
+        }
+
+        // this call somehow messes with the snapshots
+        const smc = new SourceMapConsumer(mapToUse)
+        const originalStartPosition = smc.originalPositionFor(path.node.loc.start)
+        const originalEndPosition = smc.originalPositionFor(path.node.loc.end)
+        const original = cleanSteganoTextData(path.node.value).cleaned
```

Comment:
is this a safety that you are calling cleanSteganoTextData on the user's original source code?

Reply:
I just checked and in https://github.com/concrete-utopia/utopia/blob/1d97eadf633d48cc2e92c6f0ac188be7f340ee9d/editor/src/core/workers/parser-printer/parser-printer-transpiling.ts#L68 we use `path.replaceWith` - maybe that's what we should do here instead of reassigning path.node.value too? 

also, I think we have some tests for source maps, I'm gonna dig them up!

Reply:
here! https://github.com/concrete-utopia/utopia/blob/1d97eadf633d48cc2e92c6f0ac188be7f340ee9d/editor/src/core/workers/parser-printer/parser-printer.spec.ts#L5130

Reply:
this is testing the source maps end-to-end https://github.com/concrete-utopia/utopia/blob/1d97eadf633d48cc2e92c6f0ac188be7f340ee9d/editor/src/components/canvas/ui-jsx-canvas-errors.spec.tsx#L5